osu.Game/Screens/Play/Player.cs

```diff
@@ -341,6 +359,14 @@ private void onBreakTimeChanged(ValueChangedEvent<bool> isBreakTime)
             HUDOverlay.KeyCounter.IsCounting = !isBreakTime.NewValue;
         }
 
+        private void updateOverlayActivationMode()
+        {
+            if (DrawableRuleset.HasReplayLoaded.Value)
+                OverlayActivationMode.Value = OverlayActivation.UserTriggered;
+            else
+                OverlayActivationMode.Value = gameplayOverlaysDisabled.Value && !DrawableRuleset.IsPaused.Value ? OverlayActivation.Disabled : OverlayActivation.UserTriggered;
```

Comment:
can't say I like the way this is written too much, feels like too much branching; would probably rewrite to
```csharp
bool canTriggerOverlaysDuringGameplay = DrawableRuleset.IsPaused.Value || !gameplayOverlaysDisabled.Value;
if (DrawableRuleset.HasReplayLoaded.Value || canTriggerOverlaysDuringGameplay)
    OverlayActivationMode.Value = OverlayActivation.UserTriggered;
else
    OverlayActivationMode.Value = OverlayActivation.Disabled;
```

feels more legible that way to me